File: ui/visual-component/src/main/java/org/visual/component/display/container/SplitAnimatePane.java
package org.visual.component.display.container;

import javafx.beans.property.SimpleBooleanProperty;
import javafx.scene.control.SplitPane;

public class SplitAnimatePane extends SplitPane {
  private final SimpleBooleanProperty collapse = new SimpleBooleanProperty(false);

  public Boolean getCollapse() {
    return collapse.get();
  }

  public void setCollapse(Boolean value) {
    if (value != null) {
      collapse.set(value);
    }
  }

  public void toggle() {
    collapse.set(collapse.get());
  }
}

File: ui/visual-component/src/main/kotlin/org/visual/component/container/SplitAnimatePane.kt
package org.visual.component.container

import javafx.beans.property.BooleanProperty
import javafx.beans.property.SimpleBooleanProperty
import javafx.scene.control.SplitPane
import java.io.File

class SplitAnimatePane : SplitPane() {
    private val _collapse = SimpleBooleanProperty(false)

    var collapse: Boolean?
        get() = _collapse.get()
        set(value) {
            value?.let {
                _collapse.set(value)
            }
        }
    fun toggle() {
        _collapse.set(_collapse.value)
    }
}
